### PR TITLE
Add and update Dialog component themes

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Adds new theme for DialogSection and updates some existing styles of Dialog according to design specs

**New stuff** ✨

* Added "footer" theme for DialogSection component
* Added "heading" and "footer" theme to TransferFundsDialog

**Changes** 🏗

* Updated border-radius of Dialog
* Updated padding of Dialog heading and main sections

FireShot Capture 049 - Colony - localhost

Resolves DEV-41

### CODE REVIEW

File: src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx [hunk 1]
```diff
@@ -344,7 +344,7 @@ const TransferFundsDialogForm = ({
           />
         </div>
       </DialogSection>
-      <DialogSection appearance={{ align: 'right' }}>
+      <DialogSection appearance={{ align: 'right', theme: 'footer' }}>
```
Comment (rdig) [inline src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx:347]: The confirm button needs to be wider. Currently yours is `120px`, and it needs to be `160px`.

Look up how we implemented this in `OrganizationAddDomains`: 
https://github.com/JoinColony/colonyDapp/blob/05131f8ce4c9e23793d2ada77ba7026a318f4ff2/src/modules/admin/components/Domains/OrganizationAddDomains.tsx#L70

By overwriting the component styles directly, with a value set in the css modules, using `@value`:
https://github.com/JoinColony/colonyDapp/blob/05131f8ce4c9e23793d2ada77ba7026a318f4ff2/src/modules/admin/components/Domains/OrganizationAddDomains.css#L3

**Review by rdig — CHANGES_REQUESTED** [review]
I'm good with this, however, with the exception of one, you didn't update all the modals.

Modals still needed to be updated:
- ColonyPermissionsAddDialog
- ColonyTokenEditDialog
- CreatePaymentDialog
- PermissionManagementDialog
- TokenMintDialog
- TransferFundsDialog

Make sure to update the button width on all of them

:+1:

File: src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx (hunk 1 above)
Comment (ArmandoGraterol) [reply src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx:347]: Mmmm I think it's gonna be better if we update the size of large buttons. According to the style guide, the large button is 160x44 so by updating the padding applied by the appearance, we should be fine

Comment (rdig) [reply src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialogForm.tsx:347]: Not necesarily. 

The `large` button theme is in use elsewhere as well. So if you change that, you'll modify a lot of places in the app we might not want to change.

That's why I suggested you make the change locally just for the dialog.

File: src/modules/core/components/Dialog/Dialog.css [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 .main {
```
Comment (rdig) [inline src/modules/core/components/Dialog/Dialog.css:9]: Please increase the distance between the actual modal and the `circle-close` icon

Screenshot from 2020-12-10 21-00-21

**Review by rdig — APPROVED** [review]
I'm all good with this.

But before you merge it, please increase the distance between the modal and the close button.

Nice work :+1:

**Pull request merged**